pkg/dao/repository_configs.go

```diff
@@ -372,7 +372,53 @@ func (r repositoryConfigDaoImpl) filteredDbForList(OrgID string, filteredDB *gor
 
 	if filterData.Status != "" {
 		statuses := strings.Split(filterData.Status, ",")
-		filteredDB = filteredDB.Where("status IN ?", statuses)
+
+		filteredDB = filteredDB.
+			Joins("LEFT JOIN tasks ON repository_configurations.last_snapshot_task_uuid = tasks.id").
+			Preload("Repository").
+			Preload("LastSnapshotTask")
+
+		var filterChain *gorm.DB
+		for _, status := range statuses {
+			switch status {
+			case config.StatusValid, config.StatusUnavailable, config.StatusInvalid, config.StatusPending:
+				if filterChain == nil {
+					// first where statement, so generate just it
+					filterChain = getStatusFilter(status, r.db)
+				} else {
+					// after the first, OR it with the previous statements
+					filterChain = filterChain.Or(getStatusFilter(status, r.db))
+				}
+			default:
+				log.Error().Msgf("Invalid status provided: %v", status)
+				return filteredDB
```

Comment:
i think an invalid status should probably result in a 400 or no results (probably a 400 is better), versus here it'd just return all results.

Reply:
makes sense, pushed up that fix

Reply:
woops looks like i've changed it to send a 404, let me fix that